common/gzio.cpp

```diff
@@ -1113,7 +1113,8 @@ uint32 GzioReadStream::read(void *dataPtr, uint32 dataSize) {
 
 	int32 actualRead = readAtOffset(_streamPos, (byte *)dataPtr, dataSize);
 	if (actualRead < 0) {
-		return -1;
+		_err = true;
+		return 0;
```

Comment:
Is this correct? Shouldn't that be -1? The function should return an int32 in such case

Reply:
This function is an override from `Common::ReadStream` so the return type can't be changed.  It's supposed to return the number of bytes successfully read, so I think the right thing to do if there is a read error is to flag the error (which actually happens in `readAtOffset` anyway, but, I don't think it hurts to be safe here...) and return 0.